### ISSUE TITLE
Problems creating Tables

### ISSUE THREAD

**Issue by gustavohmsilva** [yue/yue#74]
I am trying to create a simple app that only shows a table on screen using simpleTableModel with no luck, here are the attempts I've made without success and the errors that I've received:

```
local contentView = gui.Container.create()
win:setcontentview(contentView)
local tabela = gui.SimpleTableModel.create(1)
contentView:addchildview(tabela)

>>>

lua5.3: error converting arg at index 2 from yue.SimpleTableModel to yue.View
stack traceback:
        [C]: in ?
        [C]: in method 'addchildview'
        teste.lua:14: in main chunk
        [C]: in ?
```

```
local contentView = gui.Container.create()
win:setcontentview(contentView)
local tabela = gui.SimpleTableModel:create(1)
contentView:addchildview(tabela)

>>>

lua5.3: error converting arg at index 1 from table to integer
stack traceback:
        [C]: in method 'create'
        teste.lua:13: in main chunk
        [C]: in ?
```

Since the manual don't have examples, I'm having problems working with this element. Adding rows to the table don't change the error FYI...

**Comment by caguiclajmg** [comment]
`TableModel`s are just data containers and not `View`s themselves. You need to create a `Table` then use `setModel()`:

```lua
local tableModel = gui.SimpleTableModel.create(1)
-- fill table here

local table = gui.Table.create()
table:setmodel(tableModel)
```

**Comment by gustavohmsilva** [comment]
Hi @caguiclajmg, I thought this as well, but gui.Table.create() gives me error as well...

```
local contentView = gui.Container.create()
win:setcontentview(contentView)
local tableModel = gui.SimpleTableModel.create(3)
tableModel:addrow{
        'test',
        '123',
        'testing'
}
local tableView = gui.Table.create() -- this is line 20
tableView:setmodel(tableModel)
contentView:addchildview(tableView)

>>>
lua5.3: teste.lua:20: attempt to index a nil value (field 'Table')
stack traceback:
        teste.lua:20: in main chunk
        [C]: in ?

```

**Comment by caguiclajmg** [comment]
Huh that's weird, I just checked and `Table` is indeed not in the module exports.

**Comment by zcbenz** [comment]
Thanks for reporting this, I just found that I made a typo when exporting the `Table` API, so in Lua bindings `gui.TableModel` is wrongly directly to the `gui.Table`. The correct usage should be @caguiclajmg 's answer.

I'll make a fix in next release.

**Comment by gustavohmsilva** [comment]
Whoah! Did I just found a bug?! That's awesome! Not the bug I mean, but being able to help the project.

@zcbenz, I don't want to press you in any way, but how long should the next release take to get out? I was studing Yue to create a video for my Lua YT channel, but I can't do it with table this way, as it was the main reason to move from fltk4lua to yue (fltk4lua haven't implemented table views).

**Comment by caguiclajmg** [comment]
@gustavohmsilva here's a patch you can apply in the meantime

```diff
diff --git a/lua_yue/binding_gui.cc b/lua_yue/binding_gui.cc
index 0693940..9984daa 100644
--- a/lua_yue/binding_gui.cc
+++ b/lua_yue/binding_gui.cc
@@ -2117,7 +2117,7 @@ extern "C" int luaopen_yue_gui(lua::State* state) {
   BindType<nu::TableModel>(state, "TableModel");
   BindType<nu::AbstractTableModel>(state, "AbstractTableModel");
   BindType<nu::SimpleTableModel>(state, "SimpleTableModel");
-  BindType<nu::Table>(state, "TableModel");
+  BindType<nu::Table>(state, "Table");
   BindType<nu::TextEdit>(state, "TextEdit");
   BindType<nu::Tray>(state, "Tray");
 #if defined(OS_MACOSX)
```

**Comment by zcbenz** [comment]
@gustavohmsilva I'm in the middle of implementing drag and drop support, so the earliest release date would be this Sunday. You can add this code in your file to work around the bug for now:

```lua
gui.Table = gui.TableModel
```

**Comment by gustavohmsilva** [comment]
That's awesome @zcbenz , easier way ever to solve this until the new release is out. @caguiclajmg solution require compiling and several of the people who watch my videos aren't very comfortable with compiling stuff. 

Not sure if I said already, but thanks for the tool! it is incredibly easy to use, kind of like IUP from PUC-Lua, but actually way better because is Mac OS compatible :D

**Comment by zcbenz** [comment]
Fixed in 0.6.3.
